### PR TITLE
randombytes syscall : fixing types + ptr copy for returning

### PR DESCRIPTION
**Pull request by tfaoliveira**
I tested with the distribution package of libjade, and it appears to work.

### CODE REVIEW

**Comment by tfaoliveira** [comment]
I don't know how to address this line: https://github.com/jasmin-lang/jasmin/blob/cac58bdaf929eafa5f5ceb87eb264cdba9fa366a/compiler/syscall/jasmin_syscall.h#L4

**Review by bgregoir — APPROVED** [review]

File: compiler/syscall/jasmin_syscall.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #ifndef JASMIN_SYSCALL
 #define JASMIN_SYSCALL
 /* FIXME this need xlen to be Uptr */
-unsigned char * __jasmin_syscall_randombytes__(unsigned char *x, uint64_t xlen)
+uint8_t* __jasmin_syscall_randombytes__(uint8_t* x, uint64_t xlen)
 asm("__jasmin_syscall_randombytes__");
```
Comment (vbgl) [inline compiler/syscall/jasmin_syscall.h:6]: What’s the meaning of this `asm` stuff?

**Comment by vbgl** [comment]
CI: https://gitlab.com/jasmin-lang/jasmin/-/pipelines/738118512

File: compiler/syscall/jasmin_syscall.h (hunk 1 above)
Comment (tfaoliveira) [reply compiler/syscall/jasmin_syscall.h:6]: I think the goal is to ensure that the function name is preserved. Maybe this was an issue in some testing environment (but I don't precisely recall the exact reason nor wrote any notes about it).

https://gcc.gnu.org/onlinedocs/gcc/Asm-Labels.html#Asm-Labels

https://clang.llvm.org/docs/AttributeReference.html#asm

Comment (vbgl) [reply compiler/syscall/jasmin_syscall.h:6]: Thanks for the explanations & links.

**Pull request merged**